### PR TITLE
chore: update rubocop

### PR DESCRIPTION
**Pull request by dgodd**
HEAD was failing on travis, to fix that:

- update rubocop and then change code to pass the latest rubocop
- include Gemfile.lock in repo to make sure we keep the same versions for tests
- add ruby 3.0 to travis tests

### CODE REVIEW

File: lib/lightspeed/request.rb [hunk 1]
```diff
@@ -108,7 +108,7 @@ def extract_rate_limits(response)
 
     def uri
       uri = self.class.base_path + @path
-      uri += "?" + URI.encode_www_form(@params) if @params && @method == :get
+      uri += "?" + URI.encode_www_form(@params) if @params && @method == :get # rubocop:disable Style/StringConcatenation
```
Comment (robotdana) [inline lib/lightspeed/request.rb:111]: i think this is string plus? so maybe doesn't need disabling?

Comment (dgodd) [reply lib/lightspeed/request.rb:111]: I had memory that `URI.encode_www_form(@params)` did something strange. I'll double check

Comment (robotdana) [reply lib/lightspeed/request.rb:111]: I think you may be remembering e.g. URI::HTTP#+

Comment (dgodd) [reply lib/lightspeed/request.rb]: I'm wrong - I have changed the implementation - Thanks

**Review by robotdana — APPROVED** [review]

**Pull request merged**